pkg/pb/synthetic_monitoring/checks_extra_test.go

```diff
@@ -2045,3 +2049,139 @@ func TestRemoteInfoMarshalJSON(t *testing.T) {
 	actual := strings.TrimSpace(buf.String())
 	require.Equal(t, expected, actual, "JSON encoding of RemoteInfo did not match expected output")
 }
+
+// TestUsesSecretsModuleRe tests that the regular expression used to detect the use of secrets works as expected.
+//
+// When modifying the regular expression, add a case in testdata, following the following naming convention:
+//
+// * script_using_secrets_<description>.js: should be reported as using secrets
+// * script_not_using_secrets_<description>.js: should NOT be reported as using secrets
+func TestUsesSecretsModuleRe(t *testing.T) {
+	t.Parallel()
+
+	testdataDir := `testdata`
+	testdataDirFS := os.DirFS(testdataDir)
+	filenames, err := fs.Glob(testdataDirFS, "script_*.js")
+	require.NoError(t, err)
+
+	for _, filename := range filenames {
+		switch {
+		case strings.HasPrefix(filename, `script_using_secrets_`):
+			script, err := os.ReadFile(filepath.Join(testdataDir, filename))
+			require.NoError(t, err)
+			assert.Truef(t, usesSecretsModuleRe.Match(script), "%s should report true", filename)
+
+		case strings.HasPrefix(filename, `script_not_using_secrets_`):
+			script, err := os.ReadFile(filepath.Join(testdataDir, filename))
+			require.NoError(t, err)
+			assert.False(t, usesSecretsModuleRe.Match(script), "%s should report false", filename)
+
+		default: // skip
+		}
+	}
+}
+
+func TestCheckIsUsingSecrets(t *testing.T) {
```

Comment:
This test is not trying to be exhaustive (that's why the above test exists). This test is just trying to exercise the switch.